lib/factory_bot_profiler/frame.rb

```diff
@@ -1,19 +1,35 @@
 module FactoryBotProfiler
   class Frame
-    attr_accessor :name, :start, :duration, :child_time
+    attr_accessor :name, :start, :duration, :child_time, :child_count
 
     def initialize(name)
       @name = name
       @start = Time.now
-      @child_time = 0
+      @child_time = Hash.new { 0 }
+      @child_count = Hash.new { 0 }
     end
 
     def observe_child(frame)
-      @child_time += frame.duration
```

Comment:
Doesn't necessarily need to be part of this PR, but I think these counts belong in FactoryStat along with the other counts. That was true before this PR as well, but it's more noticeable to me now that we are counting more things. I thinks there's also a little bit of duplication between these two classes now.

Instead of counting things here, we could hold onto all the child frames in an array, then iterate through that array as needed in FactoryStat.

```rb
  @child_frames << frame
```

Storing all the child frames would also open up the possibility of adding custom a custom collector for building flamegraphs, etc.

Reply:
Yeah, I am leaning toward adding that. I want to figure out what the format flamegraph programs take is. I have a feeling we are going to need a little more information then time + name. it probably needs to at least know the order and when exactly in the parent process they start and end.

I think that would be awesome to get working though!